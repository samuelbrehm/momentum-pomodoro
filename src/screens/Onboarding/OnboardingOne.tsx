import React from 'react';
import { RectButton } from 'react-native-gesture-handler';
import {
  getBottomSpace,
  getStatusBarHeight,
} from 'react-native-iphone-x-helper';
import { RFValue } from 'react-native-responsive-fontsize';

import styled from 'styled-components/native';

import CompleteDesign from '../../resources/assets/complete_design.svg';
import { ContentSteps } from '../../components/ContentSteps';
import { ButtonOnboarding } from '../../components/ButtonOnboarding';

export function OnboardingOne() {
  return (
    <Container>
      <ImageTask />

      <ContentHeadingTitle>
        <HeadingTitle numberOfLines={1}>Gerencie facilmente</HeadingTitle>
        <HeadingTitle numberOfLines={1}>seu tempo e</HeadingTitle>
        <HeadingTitle numberOfLines={1}>suas tarefas com</HeadingTitle>
        <HeadingTitle numberOfLines={1}>Momentum</HeadingTitle>
      </ContentHeadingTitle>

      <ContentSteps numberStepActive={1} numberOfSteps={3} />

      <ButtonOnboarding text="PrÃ³ximo" />
    </Container>
  );
}

const Container = styled.View`
  flex: 1;
  align-items: center;
  justify-content: space-between;
  padding: 0 24px;
  background-color: ${({ theme }) => theme.colors.background};
`;

const ImageTask = styled(CompleteDesign).attrs({
  height: `${RFValue(240)}`,
})`
  margin-top: ${getStatusBarHeight() + RFValue(24)}px;
`;

const ContentHeadingTitle = styled.View`
  width: 100%;
  justify-content: space-around;
`;

const HeadingTitle = styled.Text`
  font-size: ${RFValue(25)}px;
  font-weight: bold;
  text-align: center;
  color: ${({ theme }) => theme.colors.text_primary};
  margin-bottom: ${RFValue(3)}px;
`;
