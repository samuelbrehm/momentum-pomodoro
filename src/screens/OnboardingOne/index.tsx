import React from 'react';
import { RectButton } from 'react-native-gesture-handler';
import {
  getBottomSpace,
  getStatusBarHeight,
} from 'react-native-iphone-x-helper';
import { RFValue } from 'react-native-responsive-fontsize';

import styled from 'styled-components/native';

import CompleteDesign from '../../resources/assets/complete_design.svg';

export function OnboardingOne() {
  return (
    <Container>
      <ImageTask />

      <ContentHeadingTitle>
        <HeadingTitle numberOfLines={1}>Gerencie facilmente</HeadingTitle>
        <HeadingTitle numberOfLines={1}>seu tempo e</HeadingTitle>
        <HeadingTitle numberOfLines={1}>suas tarefas com</HeadingTitle>
        <HeadingTitle numberOfLines={1}>Momentum</HeadingTitle>
      </ContentHeadingTitle>

      <ContentSteps>
        <Step isActive={true} />
        <Step isActive={false} />
        <Step isActive={false} />
      </ContentSteps>

      <NextButton>
        <TextButton>Pr√≥ximo</TextButton>
      </NextButton>
    </Container>
  );
}

const Container = styled.View`
  flex: 1;
  align-items: center;
  justify-content: space-between;
  padding: 0 24px;
  background-color: ${({ theme }) => theme.colors.background};
`;

const ImageTask = styled(CompleteDesign).attrs({
  height: `${RFValue(240)}`,
})`
  margin-top: ${getStatusBarHeight() + RFValue(24)}px;
`;

const ContentHeadingTitle = styled.View`
  width: 100%;
  justify-content: space-around;
`;

const HeadingTitle = styled.Text`
  font-size: ${RFValue(25)}px;
  font-weight: bold;
  text-align: center;
  color: ${({ theme }) => theme.colors.text_primary};
  margin-bottom: ${RFValue(3)}px;
`;

const ContentSteps = styled.View`
  flex-direction: row;
  align-items: center;
  justify-content: center;
  padding: ${RFValue(6)}px ${RFValue(24)}px;
`;

interface StepProps {
  isActive: boolean;
}

const Step = styled.View<StepProps>`
  background-color: ${({ isActive, theme }) =>
    isActive ? theme.colors.primary : theme.colors.primary_disabled};
  height: ${RFValue(12)}px;
  width: ${({ isActive }) => (isActive ? RFValue(36) : RFValue(12))}px;
  border-radius: ${RFValue(12 / 2)}px;
  margin-left: ${RFValue(6)}px;
  margin-right: ${RFValue(6)}px;
`;

const NextButton = styled(RectButton)`
  height: ${RFValue(65)}px;
  width: ${RFValue(240)}px;
  background-color: ${({ theme }) => theme.colors.primary};

  align-items: center;
  justify-content: center;

  border-radius: ${RFValue(65 / 2)}px;

  margin-bottom: ${getBottomSpace() + 24}px;
`;

const TextButton = styled.Text`
  color: ${({ theme }) => theme.colors.text_primary};
  text-align: center;
  font-size: ${RFValue(18)}px;
  font-weight: bold;
`;
