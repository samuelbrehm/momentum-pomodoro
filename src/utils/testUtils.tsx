import React from 'react';
import Theme from '../components/Theme';

import {
  cleanup,
  fireEvent,
  render as testingRender,
  waitFor,
} from '@testing-library/react-native';

export { cleanup, fireEvent, waitFor };

interface ITestWrapperProps {
  children: JSX.Element;
}

export const TestWrapper = ({ children }: ITestWrapperProps) => (
  <Theme>{children}</Theme>
);

export const render = (children: JSX.Element) =>
  testingRender(children, { wrapper: TestWrapper });
