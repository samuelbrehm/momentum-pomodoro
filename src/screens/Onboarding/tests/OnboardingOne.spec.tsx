import React from 'react';
import { render } from '../../../utils/testUtils';
import { OnboardingOne } from '../OnboardingOne';

describe('OnboardingOne', () => {
  it('should render correctly', () => {
    const { getByTestId, toJSON } = render(<OnboardingOne />);
    expect(getByTestId('onboarding-one')).toBeDefined();
    expect(getByTestId('onboarding-one')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
});
