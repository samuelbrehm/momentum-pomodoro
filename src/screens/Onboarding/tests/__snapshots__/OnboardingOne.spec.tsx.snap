// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnboardingOne should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#f0f0f0",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "space-between",
        "paddingBottom": 0,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 0,
      },
    ]
  }
  testID="onboarding-one"
>
  <SvgMock
    height="471"
    style={
      Array [
        Object {
          "marginTop": 67,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "justifyContent": "space-around",
          "width": "100%",
        },
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#030303",
            "fontSize": 49,
            "fontWeight": "bold",
            "marginBottom": 6,
            "textAlign": "center",
          },
        ]
      }
    >
      Gerencie facilmente
    </Text>
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#030303",
            "fontSize": 49,
            "fontWeight": "bold",
            "marginBottom": 6,
            "textAlign": "center",
          },
        ]
      }
    >
      seu tempo e
    </Text>
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#030303",
            "fontSize": 49,
            "fontWeight": "bold",
            "marginBottom": 6,
            "textAlign": "center",
          },
        ]
      }
    >
      suas tarefas com
    </Text>
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#030303",
            "fontSize": 49,
            "fontWeight": "bold",
            "marginBottom": 6,
            "textAlign": "center",
          },
        ]
      }
    >
      Momentum
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingBottom": 12,
          "paddingLeft": 47,
          "paddingRight": 47,
          "paddingTop": 12,
        },
      ]
    }
  >
    <View
      isActive={true}
      style={
        Array [
          Object {
            "backgroundColor": "#E8275B",
            "borderRadius": 12,
            "height": 24,
            "marginLeft": 12,
            "marginRight": 12,
            "width": 71,
          },
        ]
      }
    />
    <View
      isActive={false}
      style={
        Array [
          Object {
            "backgroundColor": "#E5BFC9",
            "borderRadius": 12,
            "height": 24,
            "marginLeft": 12,
            "marginRight": 12,
            "width": 24,
          },
        ]
      }
    />
    <View
      isActive={false}
      style={
        Array [
          Object {
            "backgroundColor": "#E5BFC9",
            "borderRadius": 12,
            "height": 24,
            "marginLeft": 12,
            "marginRight": 12,
            "width": 24,
          },
        ]
      }
    />
  </View>
  <RNGestureHandlerButton
    activeOpacity={0.105}
    collapsable={false}
    handlerTag={1}
    handlerType="NativeViewGestureHandler"
    onActiveStateChange={[Function]}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#E8275B",
        "borderRadius": 64,
        "justifyContent": "center",
        "marginBottom": 24,
        "paddingBottom": 47,
        "paddingLeft": 177,
        "paddingRight": 177,
        "paddingTop": 47,
      }
    }
    underlayColor="black"
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "black",
          "borderBottomLeftRadius": undefined,
          "borderBottomRightRadius": undefined,
          "borderRadius": 64,
          "borderTopLeftRadius": undefined,
          "borderTopRightRadius": undefined,
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#030303",
            "fontSize": 35,
            "fontWeight": "bold",
            "textAlign": "center",
          },
        ]
      }
    >
      Pr√≥ximo
    </Text>
  </RNGestureHandlerButton>
</View>
`;
